### PR TITLE
Sliceext examples

### PR DESCRIPTION
**Pull request by carols10cents**
Hi! I added some examples to some stable SliceExt methods that didn't have any.

I'm looking forward to feedback and I'm happy to change anything-- it looks like the doc conventions are still a bit in flux, based on the discussions going on in [rfc 505](https://github.com/rust-lang/rfcs/pull/505).

I was most unsure about examples for methods that return iterators over slices... I wanted to use asserts on the result of calling `.next()` like in [this permutations example](https://github.com/carols10cents/rust/blob/804c1446b3b0afd84851339d8ee2be1dca8f7713/src/libcollections/slice.rs#L608-L617), but then it gets all cluttered up with lifetime stuff... so I went with iterating and printing and mentioning what the expected printed output is like in [this chunks example](https://github.com/carols10cents/rust/blob/804c1446b3b0afd84851339d8ee2be1dca8f7713/src/libcollections/slice.rs#L297-L304)... any ideas for the best ways to do this are appreciated.

Thank you! :heart:

### CODE REVIEW

**Comment by carols10cents** [comment]
Also, I tried running `make check` but I'm running into [this issue](https://github.com/rust-lang/rust/issues/20055). I ran all the code I added manually though...

**Comment by carols10cents** [comment]
`make tips` helped me find `make check-docs`! running that now :)

**Comment by carols10cents** [comment]
Yay that worked! Let me know if I should do anything else :)

File: src/libcollections/slice.rs [hunk 1]
```diff
@@ -188,15 +188,36 @@ pub trait SliceExt {
...
     ///
     /// Panics if `mid > len`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let v = [10i32, 40, 30, 20, 50];
+    /// let (v1, v2) = v.split_at(2);
+    /// assert!(v1 == [10, 40]);
+    /// assert!(v2 == [30, 20, 50]);
```
Comment (steveklabnik) [inline src/libcollections/slice.rs]: I think these should be `assert_eq`

File: src/libcollections/slice.rs [hunk 2]
```diff
@@ -188,15 +188,36 @@ pub trait SliceExt {
     /// indices from `[mid, len)` (excluding the index `len` itself).
     ///
     /// Panics if `mid > len`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let v = [10i32, 40, 30, 20, 50];
```
Comment (alexcrichton) [inline src/libcollections/slice.rs]: I think in all of these examples you may be able to leave out the `i32` suffix entirely

**Comment by carols10cents** [comment]
All fixed! Thank you for the feedback-- I changed to `assert_eq` and was able to remove the `i32` from all but the empty slices :)

r? @steveklabnik @alexcrichton

**Comment by alexcrichton** [comment]
@bors: r+ edb2d8d rollup

thanks!

**Pull request merged**